### PR TITLE
Deprecates IO data type and related type classes in favor of Arrow Fx Coroutines

### PR DESCRIPTION
**Pull request by raulraja**
## Description
Deprecates IO data type and related type classes for 0.12.0

I adapted the current docs to remove all references to IO and its type classes. We still need to come up with better docs and structure for all the libs that would come in other PRs. 

## Todos
- [x] Documentation

## Related PRs
https://github.com/arrow-kt/arrow-core/pull/275

### CODE REVIEW

**Review by i-walker — APPROVED** [review]
Thanks so much @raulraja for this 🙌

File: arrow-docs/docs/landscape/fx-home-code.md [hunk 1]
```diff
@@ -1,24 +1,34 @@
...
-}.suspended()
-//sampleEnd
+suspend fun main() {
+    //sampleStart
+    //maps each function to `::employee` in parallel
+    val audrey = parMapN({ "Audrey" }, { company("Arrow") }, ::employee) 
+    val pepe   = parMapN({  "Pepe"  }, { company("Arrow") }, ::employee)
+    val candidates = listOf(audrey, pepe)
+    val employees = candidates.parTraverse(::hire) //hires in parallel
+    //sampleEnd
```
Comment (nomisRev) [inline arrow-docs/docs/landscape/fx-home-code.md:31]: :tada:

**Review by nomisRev — APPROVED** [review]
Awesome ❤️

File: arrow-docs/docs/fx/async/README.md [hunk 2]
```diff
@@ -329,15 +329,8 @@ Strange, you might've expected a `Coroutine` type but a `Coroutine` is represent
...
-
-Both Arrow Fx Coroutines & KotlinX Coroutines independently offer an implementation for Kotlin's coroutine system.
-
-As explained in the document above, Arrow Fx Coroutines offers a battery-included functional IO with cancellation support.
-Where KotlinX Coroutines offers an implementation that offers light-weight futures with cancellation support.
-
 ## Integrating with third-party libraries
 
-Arrow Fx integrates with the Arrow Fx IO runtime, Rx2, Reactor framework, and any library that can model effectful async/concurrent computations as `suspend`.
+Arrow Fx integrates with KotlinX Coroutines Fx, Reactor framework, and any library that can model effectful async/concurrent computations as `suspend`.
```
Comment (1Jajen1) [inline arrow-docs/docs/fx/async/README.md:334]:
```suggestion
Arrow Fx integrates with KotlinX Coroutines, Reactor framework, and any library that can model effectful async/concurrent computations as `suspend`.
```

**Review by 1Jajen1 — APPROVED** [review]
:tada:

**Pull request merged**